indexing-service/src/main/java/io/druid/indexing/overlord/http/OverlordResource.java

```diff
@@ -372,7 +454,36 @@ public Response getRunningTasks()
           @Override
           public Collection<? extends TaskRunnerWorkItem> apply(TaskRunner taskRunner)
           {
-            return taskRunner.getRunningTasks();
+            if (authConfig.isEnabled()) {
```

Comment:
this and the change in getPendingTasks() are same, can you refactor them into a private helper method?